### DISCUSSION TITLE
does anyone know how to keep the formatting of prompts consistent across different topic?

### DISCUSSION THREAD

**Discussion by hiqsociety** [ggml-org/llama.cpp#3217] [Q&A] [answered]
e.g. i would like to generate wikipedia style article. how to prompt for consistency across all topic / titles?

below is my prompt and i know the seed value but i can never find the right seed to generate article like wikipedia. does anyone know what trick i can get wikipedia like comprehensive and detailed article? e.g. more than 2400 words too etc.

```
/main -m models/llama-2-7b-lora-assemble.Q4_K_M.gguf -ngl 35 -c 3620 -n 12288 -p "Detailed encyclopedia-style article titled 'elon musk' with a minimum of 2400 words. The content should be in English and formatted in markdown. Ensure accuracy, avoid speculation. Structured with headings, an intro, and conclusion. Include inline citations, external/internal links (excluding images), and the markdown reference link format `This is [an example][id] reference-style link; [id]: http://example.com/ \"Optional Title Here\"`. Integrate advanced markdown elements and a table of contents where appropriate." -e -t 1
```

**Comment by KerfuffleV2** [comment, marked as answer]
Everything else aside:

> `-c 3620 -n 12288`

That can't work. `-c` sets a context size of 3,620 tokens. Both the prompt and any generated tokens need to fit in that (unless you're using --keep but I wouldn't really recommend it and even when it works you're not likely to get coherent output too far past double the context size).

Since `-c 3620`, `-n` needs to be a lower value. Your model is LLaMA 2 so it probably supports up to `-c 4096`. You can also possibly look into using RoPE tricks to be able to set `-c` to a higher value but expecting 12k tokens worth of coherent output from a 7B model is pretty optimistic.
